ports/esp32/machine_wdt.c

```diff
@@ -42,17 +43,32 @@ typedef struct _machine_wdt_obj_t {
 STATIC machine_wdt_obj_t wdt_default = {{&machine_wdt_type}};
 
 STATIC mp_obj_t machine_wdt_make_new(const mp_obj_type_t *type_in, size_t n_args, size_t n_kw, const mp_obj_t *args) {
-    mp_arg_check_num(n_args, n_kw, 0, 1, false);
+    mp_arg_check_num(n_args, n_kw, 0, 3, false);
 
     mp_int_t id = 0;
+    mp_int_t timeout = 10;
+    bool panic = true;
+
     if (n_args > 0) {
         id = mp_obj_get_int(args[0]);
+        timeout = mp_obj_get_int(args[1]);
     }
```

Comment:
I don't see any place where you read the value of `args[2]`. It looks like this will always panic on timeout if you use `id==1`.